packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -736,6 +736,10 @@ class BaseSynchronizedData:
     This is the relevant data constructed and replicated by the agents.
     """
 
+    # Keys always set by default
+    # TODO: round_count and period_count need to be guaranteed to be synchronized too.
```

Comment:
@Adamantios thoughts on this in context of safe deployment removal?

Reply:
We can add it here once you have changed the functionality on the new PRs

Reply:
I'd replace the todo with this comment

Reply:
Done in c8731c93530f42dff8522e15be7b1062b38a6a57.

Reply:
Actually I think it's a good idea to move them in `SynchronizedData`, they are a bit messy atm. Reset index is both defined on `AbciApp` and `AbciAppDB`. The one from `AbciApp` is used in generating the root hash. Where as the other one is used in `reset_pause_abci`. In the latter case its sent as a payload, but we never update the db. I think it would make sense to update the db with majority of the `reset_index`. Also, I don't think it makes sense to assume that all agents have the same `len(self._data)` in `AbciAppDB`, especially with rejoins (recovery) in mind.

Reply:
One reason why we want to have `round_count` in `SynchronizedData` is becuase of background behaviours. It would help us avoid possible race conditions with concurrent rounds. We have yet to see this happen, but I think there edge cases when it could happen.